lib/rucio/core/replica_sorter.py

```diff
@@ -158,6 +158,59 @@ def __get_distance(se1, client_location, ignore_error):
     return cache_val
 
 
+custom_distance_table = {}
+custom_distance_download_time = None
+
+
+def __download_custom_distance_table() -> None:
+    global custom_distance_table, custom_distance_download_time
+    # if custom distance table has expired or not been downloaded, download it now
+    db_expire_delay = timedelta(days=config_get_int('core', 'custom_distance_expire_delay', raise_exception=False, default=30))
+    if custom_distance_download_time is None or custom_distance_download_time < datetime.now() - db_expire_delay:
+        download_url = config_get('core', 'custom_distance_download_url', raise_exception=False, default=None)
+        if download_url is None:
+            raise Exception('Cannot download custom distance table: no URL provided')
+        result = requests.get(download_url, stream=True, verify=False)
+        if result and result.status_code in [200, ]:
+            new_custom_distance_table = {}
+            # parse distance table consisting of site name, RSE name and distance on each line
+            lines = result.text.split("\n")
+            for line in lines:
+                if line.strip() == "":
+                    # ignore blank lines
+                    continue
+                bits = line.split(",")
+                # DUNE's CSV file has additional fields after the first 3 so just ignore those
+                site = bits[0].strip()
+                rse = bits[1].strip()
+                distance = float(bits[2].strip())
+                if distance < 0.0 or distance > 1.0:
+                    raise Exception('Distances in custom distance table must be in range 0-1')
+                if site not in new_custom_distance_table:
+                    new_custom_distance_table[site] = {}
+                new_custom_distance_table[site][rse] = distance
+            custom_distance_download_time = datetime.now()
+            custom_distance_table = new_custom_distance_table
+        else:
+            raise Exception('Cannot download custom distance table: %s, Code: %s, Error: %s' % (download_url,
+                                                                                                result.status_code,
+                                                                                                result.text))
+
+
+def __get_distance_custom(rse: Union[tuple, str], client_location: dict) -> float:
+    # get RSE name out of tuple if necessary
```

Comment:
I think this should be re-written to use the memcache REGION which is also used above. It is better than to rely on these module-loaded dictionaries.

Reply:
I've modified the code to store the data in the memcache REGION.